# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: 48eaa84c206d
# Full Hash: 48eaa84c206dbdda637c57b55ddbd07e975ba103
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-23 16:15:06
# Regressor Bug: 1637500
# File Overlap Count: 2
# Description:
#   Bug 1637500 - P2. Rename methods as they are not always dealing with "threads". r=froydnj
#   
#   Before P1, GetCurrentThreadSerialEventTarget would have always returned the same data as NS_GetCurrentThread, making the comment incorrect Now it will properly return the running TaskQueue if any.
#   
#   This change of name more clearly exposes what they are doing, as we aren't always dealing with threads directly; but a nsISerialEventTarget
# ==============================================================================

diff -r 00687d06a8a6 -r 48eaa84c206d dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Tue Jun 23 05:02:39 2020 +0000
+++ b/dom/ipc/ContentParent.cpp	Tue Jun 23 05:05:36 2020 +0000
@@ -1079,7 +1079,7 @@
   // against whoever launched it (and whoever else is already racing). Once
   // the race is complete, the winner will finish the initialization.
   return mSubprocess->WhenProcessHandleReady()->Then(
-      GetCurrentThreadSerialEventTarget(), __func__,
+      GetCurrentSerialEventTarget(), __func__,
       // On resolve.
       [self = RefPtr{this}, aPriority]() {
         if (self->IsLaunching()) {
@@ -2435,7 +2435,7 @@
   mLaunchYieldTS = TimeStamp::Now();
 
   return ready->Then(
-      GetCurrentThreadSerialEventTarget(), __func__,
+      GetCurrentSerialEventTarget(), __func__,
       [self, aInitialPriority](
           const ProcessHandlePromise::ResolveOrRejectValue& aValue) {
         if (aValue.IsResolve() &&
@@ -3875,7 +3875,7 @@
     InitStreamFilterResolver&& aResolver) {
   extensions::StreamFilterParent::Create(this, aChannelId, aAddonId)
       ->Then(
-          GetCurrentThreadSerialEventTarget(), __func__,
+          GetCurrentSerialEventTarget(), __func__,
           [aResolver](mozilla::ipc::Endpoint<PStreamFilterChild>&& aEndpoint) {
             aResolver(std::move(aEndpoint));
           },
@@ -6057,7 +6057,7 @@
       aTopLevelWindowId, aParentContext.get_canonical(), aTrackingPrincipal,
       aTrackingOrigin, aAllowMode)
       ->Then(
-          GetCurrentThreadSerialEventTarget(), __func__,
+          GetCurrentSerialEventTarget(), __func__,
           [aResolver = std::move(aResolver)](
               ContentBlocking::ParentAccessGrantPromise::ResolveOrRejectValue&&
                   aValue) {
@@ -6082,7 +6082,7 @@
   ContentBlocking::CompleteAllowAccessFor(
       aParentContext.get_canonical(), aTopLevelWindowId, aTrackingPrincipal,
       aTrackingOrigin, aCookieBehavior, aReason, nullptr)
-      ->Then(GetCurrentThreadSerialEventTarget(), __func__,
+      ->Then(GetCurrentSerialEventTarget(), __func__,
              [aResolver = std::move(aResolver)](
                  ContentBlocking::StorageAccessPermissionGrantPromise::
                      ResolveOrRejectValue&& aValue) {